### ISSUE TITLE
Small Issue with  EdiValidator.Validate(object message) and XsdCache

### ISSUE THREAD

**Issue by jonreis** [DonZoeggerle/ediFabric#53]
I think that       

 XmlSchemaSet schemas = EdiValidator.XsdCache.GetOrAdd(message.GetType().FullName, EdiValidator.NewSchemaSet(this._xsdStream(new MessageContext(message)), source.Root.Name.Namespace.NamespaceName));

needs to be

 XmlSchemaSet schemas = EdiValidator.XsdCache.GetOrAdd(message.GetType().FullName,  key=>EdiValidator.NewSchemaSet(this._xsdStream(new MessageContext(message)), source.Root.Name.Namespace.NamespaceName));

Otherwise, the Func<MessageContext, Stream> gets called even if it isn't needed, which defeats the purpose of the Func.

in other words, use ConcurrentDictionary:

public TValue GetOrAdd(
	TKey key,
	Func<TKey, TValue> valueFactory
)

not

public TValue GetOrAdd(
	TKey key,
	TValue value
)

**Comment by DonZoeggerle** [comment]
Well spotted Jon, thank you for the input. This has been fixed in 7.3.1
